pkg/util/helm_helpers.go

```diff
@@ -347,6 +363,29 @@ func ReleaseExists(releaseName, namespace, kubeConfig string) bool {
 	return true
 }
 
+func mergeExtraFilesToConfig(ch *chart.Chart, vals map[string]interface{}, extraFiles []string) error {
+	for _, fileName := range extraFiles {
+		found := false
+		for _, chartFile := range ch.Files {
+			if fileName == chartFile.Name {
+				found = true
+				patch := make(map[string]interface{})
+				if err := yaml.Unmarshal(chartFile.Data, &patch); err != nil {
+					return err
+				}
+				if err := mergo.Merge(&vals, patch, mergo.WithOverride); err != nil {
```

Comment:
I'm not sure why the Helm team wrote their own function to do this merging: https://github.com/helm/helm/blob/9b42702a4bced339ff424a78ad68dd6be6e1a80a/pkg/cli/values/options.go#L88

Reply:
fyi this is how they are doing their merging: https://github.com/helm/helm/blob/9b42702a4bced339ff424a78ad68dd6be6e1a80a/pkg/cli/values/options.go#L45    :)